### PR TITLE
Extend boto_iam with support for managed policies

### PR DESCRIPTION
**Pull request by kraney**
Managed policies are a relatively new feature of IAM. This change adds support for them to boto_iam and boto_iam_role.

This adds create, delete, and related functionality for customer managed policies themselves. In additional, the states for user, group, and role have all been updated to allow attachment of both customer and AWS managed policies.

Given only a policy name, this assumes a customer managed policy is what's desired and generates an ARN for that for the config. If you want AWS managed policies, you have to use their full ARN.

Users of boto_iam for users, groups, and roles previously should note that the previous behavior was to ignore managed policies, so if you attached some manually, they would remain undisturbed. Now they're managed, so if you haven't specified them in your SLS, any that were manually attached will be detached when you apply the state.

Along with this change, the missing "group_absent" state was also added, to make group management symmetrical.

### CODE REVIEW

File: salt/modules/boto_iam.py [hunk 1]
```diff
@@ -41,11 +41,13 @@
 from __future__ import absolute_import
 import logging
 import json
+import urllib
```
Comment (cachedout) [inline salt/modules/boto_iam.py]: We may want to make this `salt.ext.moves.urllib.parse`.

**Comment by cachedout** [comment]
@kraney A couple lint errors in here as well that should be easy to clean up, if you wouldn't mind having a quick look.

**Comment by cachedout** [comment]
@kraney One more lint, please. Could be a typo?

**Comment by rallytime** [comment]
Hi @kraney  - It looks like there are a couple of tests that need to be addressed. Would you mind taking a look there? Let us know if you need some assistance and we can lend a hand!

**Comment by kraney** [comment]
Will fix. Sorry

**Comment by cachedout** [comment]
Merged! Excellent work as always, @kraney

File: salt/modules/boto_iam.py [hunk 2]
```diff
@@ -46,6 +46,7 @@
 # Import salt libs
 import salt.utils.compat
 import salt.utils.odict as odict
+import salt.utils.boto
```
Comment (ryan-lane) [inline salt/modules/boto_iam.py]: Sigh. I really wish PRs would stop getting merged with this. It's not backwards compatible and it doesn't make it possible to override the modules in core. It's recommended to use __utils__['boto.function'] for accessing utils.

**Pull request merged**